pom.xml

```diff
@@ -64,11 +64,12 @@
     <properties>
         <revision>2.14</revision>
         <changelist>-SNAPSHOT</changelist>
-        <jenkins.version>2.121.1</jenkins.version>
+        <jenkins.version>2.138.4</jenkins.version>
         <java.level>8</java.level>
+        <useBeta>true</useBeta>
         <workflow-cps-plugin.version>2.69</workflow-cps-plugin.version>
-        <git-plugin.version>3.6.4</git-plugin.version>
-        <scm-api-plugin.version>2.2.7</scm-api-plugin.version>
+        <git-plugin.version>4.0.0-rc</git-plugin.version>
```

Comment:
PR looks good, but I want to wait for a stable release of `git` 4.x before merging the change (Commenting so I don't forget about this later). What would happen if we merged this as-is and released to the non-experimental update center? Broken instances for everyone because they can't satisfy the `git` dependency?

Reply:
`git` is only a `test`-scoped dep, but `git-client` is a problem because it is in `dependencies` as `compile` scope:

```
Plugin-Dependencies: workflow-cps:2.69,workflow-scm-step:2.4,workflow-step-api:2.19,workflow-support:3.3,cloudbees-folder:6.1.0,credentials:2.1.14,git-client:3.0.0-rc,git-server:1.7,scm-api:2.3.0,script-security:1.58,structs:1.18
```

I suspect it is only there to satisfy `RequireUpperBoundsDeps`, in which case it would better be moved to `dependencyManagement` and then would not appear in the release at all. Would that suffice from your PoV?

Reply:
Oops, I totally missed `git` being test-scope. Yeah if moving `git-client` into `dependencyManagement` gets it out of compile scope but tests still pass then I'd be fine to merge the change at that point.

Reply:
OK, will try that.

Reply:
Ok, merged this a bit prematurely because I did not actually run `mvn dependency:tree`. This library has a compile-scope dependency on git-server which itself has a compile-scope dependency on git-client, so moving `git-client` to `dependencyManagement` does not fix the problem, and so the plugin currently has an experimental release as a compile-scope dependency. I'm not sure to what extent `git-server` uses `git-client` or if there is an easy fix for this. Maybe safest to revert the merge for now?

Reply:
> moving `git-client` to `dependencyManagement` does not fix the problem

It does because it does not have a _direct_ dependency on `git-client`, so it does not appear in the manifest.

```sh
mvn -DskipTests clean package
unzip -p target/workflow-cps-global-lib.hpi META-INF/MANIFEST.MF | perl -p -0777 -e 's/\r?\n //g' | fgrep Plugin-Dependencies
```

⇒

```
Plugin-Dependencies: workflow-cps:2.69,workflow-scm-step:2.4,workflow-step-api:2.19,workflow-support:3.3,cloudbees-folder:6.1.0,credentials:2.1.14,git-server:1.7,scm-api:2.3.0,script-security:1.58,structs:1.18
```

Reply:
Ok makes sense, in an actual instance the version of git-client required by the plugin will be whatever git-server 1.7 requires because of how Jenkins resolves dependencies.

Reply:
Rght.